_extensions/acronyms/acronyms.lua

```diff
@@ -65,21 +66,42 @@ function Acronym:new(object)
         raiseAcronymCreationError(object)
     end
 
-    -- If the key is not set, we want to use the shortname instead.
-    -- (Most of the time, the key is the shortname in lower case anyway...)
-    object.key = object.key or object.shortname
+    -- Enforce explicit key when markdown parsing for shortname is enabled.
+    if Helpers.contains_markdown(object.shortname) then
+        if object.key == nil then
+            quarto.log.error("[acronyms] Each acronym must provide an explicit `key` when using markdown in shortname.")
+            assert(false)
+        end
+    else
+        -- Legacy fallback.
+        object.key = object.key or Helpers.inlines_to_string(object.shortname)
+    end
 
-    -- If the plural forms are not set, we construct sane defaults instead.
+    -- Track whether the user explicitly provided plural forms before we add defaults.
+    local explicit_plural_short = false
+    local explicit_plural_long  = false
+    if object.plural then
+        if object.plural.shortname ~= nil then explicit_plural_short = true end
+        if object.plural.longname  ~= nil then explicit_plural_long  = true end
+    end
```

Comment:
We can simplify this (easier to read)

```suggestion
    local explicit_plural_short = object.plural and object.plural.shortname ~= nil
    local explicit_plural_long  = object.plural and object.plural.longname ~= nil
```